torch/_dynamo/eval_frame.py

```diff
@@ -563,6 +565,27 @@ def call_compiled_autograd():
 
             self.enter_exit_hooks.append(call_compiled_autograd)
 
+    def __call__(self, fn):
+        @functools.wraps(fn)
+        def _fn(*args, **kwargs):
+            if (
+                self.warmup_count < torch._dynamo.config.warmup_runs
+                # NOTE: Compiled Autograd warmup is handled by itself, not here.
+                and not torch._dynamo.compiled_autograd.in_compiled_autograd_region
+            ):
+                inner_fn = self.disable_context(fn)
+                out = inner_fn(*args, **kwargs)
+                self.warmup_count += 1
+                log.debug("OptimizeContext warmup run done: %d", self.warmup_count)
+                return out
+            else:
+                return super(OptimizeContext, self).__call__(fn)(*args, **kwargs)
+
+        # Save the function pointer to find the original callable while nesting
+        # of decorators.
+        _fn._torchdynamo_orig_callable = fn  # type: ignore[attr-defined]
+        return _fn
```

Comment:
I think my changes here is too hacky (and causes many test failures), and it requires a more knowledgable person here to do it more properly taking existing Dynamo assumptions into consideration. cc. @anijain2305